File: web/client/src/app/research/models/template.ts

import { Org } from 'src/app/shared/models/org';
import { TemplateSection } from './template-section';


export interface Template {
    id: number;
    name: string;
    description: string;
    minEstimatedValue: number;
    maxEstimatedValue: number;
    isActive: boolean;
    bidTypes: number[];
    orgs: Org[];
    serviceTypes: number[];
    sourceTypes: number[];
    sections: TemplateSection[];
}

File: web/api/afmr.model/Research/Template.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace afmr.model.Research
{
    public class Template
    {
        public int Id { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public decimal MinEstimatedValue { get; set; }

        public decimal MaxEstimatedValue { get; set; }

        public bool IsActive { get; set; }

        public IEnumerable<BidType> BidTypes { get; set; }

        public IEnumerable<Org> Orgs { get; set; }

        public IEnumerable<ServiceType> ServiceTypes { get; set; }

        public IEnumerable<SourceType> SourceTypes { get; set; }

        public IEnumerable<TemplateSection> Sections { get; set; }
    }
}
